kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/lookup/GecEntryHelperServiceImpl.java

```diff
@@ -1,54 +1,151 @@
 import org.kuali.rice.krad.util.KRADConstants;
 import org.kuali.rice.krad.util.UrlFactory;
 
-import edu.arizona.kfs.gl.businessobject.EntryLiteBo;
+import edu.arizona.kfs.gl.businessobject.GecEntry;
+import edu.arizona.kfs.gl.businessobject.lite.AccountingLineLiteBo;
+import edu.arizona.kfs.gl.businessobject.lite.DocumentHeaderLiteBo;
+import edu.arizona.kfs.gl.businessobject.lite.DocumentTypeLiteBo;
 import edu.arizona.kfs.sys.KFSConstants;
 import edu.arizona.kfs.sys.KFSPropertyConstants;
 
 @SuppressWarnings("deprecation")
-public class EntryLiteBoHelperServiceImpl {
+public class GecEntryHelperServiceImpl {
 
-    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(EntryLiteBoHelperServiceImpl.class);
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(GecEntryHelperServiceImpl.class);
+
+    private static transient volatile BusinessObjectService businessObjectService;
+
+    private static BusinessObjectService getBusinessObjectService() {
+        if (businessObjectService == null) {
+            businessObjectService = SpringContext.getBean(BusinessObjectService.class);
+        }
+        return businessObjectService;
+    }
+
+    public static String findGecDocumentNumber(String documentNumber) {
+        LOG.debug("Searching for GEC Document Number of " + documentNumber);
+        Map<String, String> fieldValues = new HashMap<String, String>();
+        fieldValues.put(KFSPropertyConstants.REFERENCE_NUMBER, documentNumber);
+        Collection<AccountingLineLiteBo> accountingLineList = getBusinessObjectService().findMatching(AccountingLineLiteBo.class, fieldValues);
+
+        LOG.debug("Checking each accounting line that uses " + documentNumber + " as the reference Number.");
+        for (AccountingLineLiteBo accountingLine : accountingLineList) {
+            String gecDocumentNumber = accountingLine.getDocumentNumber();
+            LOG.debug("Checking to see if " + gecDocumentNumber + " is a GEC.");
+            DocumentHeaderLiteBo docHeader = getBusinessObjectService().findBySinglePrimaryKey(DocumentHeaderLiteBo.class, gecDocumentNumber);
+            if (docHeader != null) {
+                String docTypeId = docHeader.getDocumentTypeId();
+                DocumentTypeLiteBo docType = getBusinessObjectService().findBySinglePrimaryKey(DocumentTypeLiteBo.class, docTypeId);
+                String documentTypeName = docType.getName();
+                if (documentTypeName.equals(KFSConstants.FinancialDocumentTypeCodes.GENERAL_ERROR_CORRECTION)) {
+                    LOG.debug("GEC Document Number " + gecDocumentNumber + " found for Document Number " + documentNumber + ".");
+                    return gecDocumentNumber;
+                }
+            }
+        }
+        LOG.debug("No GEC Document Number found for Document Number " + documentNumber + ".");
+        return KFSConstants.EMPTY_STRING;

```

Comment:
This looks to be problematic, since now you *must* do StringUtils.isBlank() everywhere on this field when checking validity, and a quick search shows, for instance, that GecEntryLookupAction#filterResults():150 is doing a straight null check. Is there a specific reason you did not go with returning null?